core/formats/formats_10.cpp

```diff
@@ -1975,9 +2047,37 @@ class doc_iterator : public doc_iterator_base<IteratorTraits, FieldTraits> {
                   [](doc_id_t doc) { return !doc_limits::valid(doc); }));
   }
 
+  void WandPrepare(const term_meta& meta, const index_input* doc_in,
+                   const index_input* pos_in, const index_input* pay_in,
+                   const ScoreFunctionFactory& factory, const Scorer& scorer,
+                   byte_type wand_index) {
+    prepare(meta, doc_in, pos_in, pay_in, wand_index);
+    if (meta.docs_count > FieldTraits::block_size()) {
+      return;
+    }
+    auto& threshold = std::get<score_threshold>(attrs_);
+    auto ctx = scorer.prepare_wand_source();
+    auto func = factory(*ctx, scorer);
+    const bool is_full_block = meta.docs_count == FieldTraits::block_size();
+    const auto was = this->doc_in_->file_pointer();
```

Comment:
virtual call is needed only in case of full block

Reply:
Fixed